### DISCUSSION TITLE
Should Linq2Acad.Tests work?

### DISCUSSION THREAD

**Discussion by stenci** [wtertinek/Linq2Acad#44] [Q&A] [answered]
I am able to build the `Linq2Acad` solution, but I can't build the `Linq2Acad.Tests` one.

One of the errors is that `db.DBVisualStyles` does not exist:
![image](https://github.com/wtertinek/Linq2Acad/assets/5955495/2788c2f8-f2ea-4f95-a2e2-abcb47ac937c)

**Comment by wtertinek** [comment]
I forgot about this solution. No, it actually doesn't work. It is based on an AutoCAD Test Runner project I once started to implemented but never really finished.

**Reply by wtertinek** [reply, marked as answer]
Well, if the test runner is set up corretly it would work, but the test runner setup is not easy to do.

**Reply by stenci** [reply]
I read that the AutoCAD CoreConsole has limitations with the entity selection and more, and I like to include different kinds of selection in my tests.

I am experimenting with the approach described [here](https://forums.autodesk.com/t5/net/how-do-you-test-commands/m-p/12551471/highlight/true#M80535): I start an AutoCAD session for every test script, then analyze the log file generated by the script to report errors to the test suite. The cool thing is that I can add tests without writing them in C#. I add a folder with an .scr script file and some drawings, and I have added one test to the test suite.

So far the drawing comparer is only working with a handful of entities in the model space. Soon I will test commands that create nested blocks and I will update the comparer to recursively compare all the blocks. And if it looks decent I might create a public repository.
